astropy/io/ascii/tests/test_read.py

```diff
@@ -1244,3 +1246,42 @@ def text_aastex_no_trailing_backslash():
     assert dat.colnames == ['a', 'b', 'c']
     assert np.all(dat['a'] == ['1', r'3\%'])
     assert np.all(dat['c'] == ['c', 'e'])
+
+
+@pytest.mark.skipif('six.PY2')
+@pytest.mark.parametrize('encoding', ['utf8', 'latin1', 'cp1252'])
+def test_read_with_encoding(tmpdir, encoding):
+    data = {
+        'commented_header': u'# à b è \n 1 2 héllo',
+        'csv': u'à,b,è\n1,2,héllo'
+    }
+
+    testfile = str(tmpdir.join('test.txt'))
+    for fmt, content in data.items():
+        with open(testfile, 'w', encoding=encoding) as f:
+            f.write(content)
+
+        table = ascii.read(testfile, encoding=encoding)
+        assert table.pformat() == [' à   b    è  ',
+                                   '--- --- -----',
+                                   '  1   2 héllo']
+
+        table = ascii.read(testfile, format=fmt, fast_reader=False,
```

Comment:
Add a loop here `for guess in (True, False):` to explicitly include the no-guessing case.